msal/application.py

```diff
@@ -506,11 +516,12 @@ def __init__(
                 self.http_client,
                 validate_authority=validate_authority,
                 instance_discovery=self._instance_discovery,
+                oidc_authority_url=oidc_authority,
                 )
         except ValueError:  # Those are explicit authority validation errors
             raise
         except Exception:  # The rest are typically connection errors
-            if validate_authority and azure_region:
+            if validate_authority and azure_region and not oidc_authority:
```

Comment:
I am not sure I fully understand this exception block, but am I right to understand that instance discovery will not be performed for oidc_authority?

Reply:
Never mind, I saw the tests.